src/test/java/uk/gov/hmcts/juror/api/moj/service/report/YieldPerformanceReportServiceImplTest.java

```diff
@@ -0,0 +1,196 @@
+    class YieldPerformanceReportTests {
+
+        @Test
+        @SneakyThrows
+        void viewYieldPerformanceReportAllCourtsNoResultsAndValidHeadings() {
+
+            final LocalDate reportFromDate = LocalDate.of(2024, 4, 20);
+            final LocalDate reportToDate = LocalDate.of(2024, 5, 13);
+
+            YieldPerformanceReportRequest yieldPerformanceReportRequest = YieldPerformanceReportRequest.builder()
+                .fromDate(reportFromDate)
+                .toDate(reportToDate)
+                .allCourts(true)
+                .build();
+
+            when(courtQueriesRepository.getAllCourtLocCodes(false))
+                .thenReturn(List.of("415", "416"));
+
+            when(poolCommentRepository.findPoolCommentsForLocationsAndDates(List.of("415", "416"), reportFromDate,
+                reportToDate))
+                .thenReturn(List.of());
+
+            when(jurorPoolRepository.getYieldPerformanceReportStats("415,416", reportFromDate, reportToDate))
+                .thenReturn(List.of());
+
+            YieldPerformanceReportResponse response = yieldPerformanceReportService.viewYieldPerformanceReport(
+                yieldPerformanceReportRequest
+            );
+
+            assertThat(response.getHeadings()).isNotNull();
+            Map<String, AbstractReportResponse.DataTypeValue> headings = response.getHeadings();
+
+            validateReportHeadings(headings);
+
+            AbstractReportResponse.DataTypeValue timeCreated = headings.get("time_created");
+            assertThat(timeCreated.getDisplayName()).isEqualTo("Time created");
+            assertThat(timeCreated.getDataType()).isEqualTo("LocalDateTime");
+            LocalDateTime createdTime = LocalDateTime.parse((String) timeCreated.getValue(),
+                DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+            assertThat(createdTime).isCloseTo(LocalDateTime.now(), within(10, ChronoUnit.SECONDS))
+                .as("Creation time should be correct");
+
+            assertThat(response.getTableData()).isNotNull();
+            YieldPerformanceReportResponse.TableData tableData = response.getTableData();
+            assertThat(tableData.getHeadings()).isNotNull();
+            Assertions.assertThat(tableData.getHeadings()).hasSize(6);
+
+            validateTableHeadings(tableData);
+
+            assertThat(tableData.getData()).isNotNull();
+            assertThat(tableData.getData().size()).isZero();
+
+            verify(courtQueriesRepository, times(1)).getAllCourtLocCodes(false);
+            verify(poolCommentRepository, times(1)).findPoolCommentsForLocationsAndDates(List.of("415", "416"),
+                reportFromDate, reportToDate);
+            verify(jurorPoolRepository, times(1)).getYieldPerformanceReportStats("415,416",
+                reportFromDate, reportToDate);
+
+        }
+
+        private void validateTableHeadings(YieldPerformanceReportResponse.TableData tableData) {
```

Comment:
Method name should be called assertTableHeadings (allows PMD to auto detect it as an assertion)

Reply:
updated